enginetest/queries/priv_auth_queries.go

```diff
@@ -952,6 +952,105 @@ var UserPrivTests = []UserPrivilegeTest{
 			},
 		},
 	},
+	{
+		Name: "show user with no grants",
+		SetUpScript: []string{
+			"CREATE USER tester@localhost;",
+		},
+		Assertions: []UserPrivilegeTestAssertion{
+			{
+				User:  "root",
+				Host:  "localhost",
+				Query: "SHOW GRANTS FOR tester@localhost;",
+				Expected: []sql.Row{
+					{"GRANT USAGE ON *.* TO `tester`@`localhost`"},
+				},
+			},
+		},
+	},
+	{
+		Name: "show grants with multiple global grants",
+		SetUpScript: []string{
+			"CREATE USER tester@localhost;",
+			"GRANT SELECT ON *.* to tester@localhost",
+			"GRANT INSERT ON *.* to tester@localhost",
+		},
+		Assertions: []UserPrivilegeTestAssertion{
+			{
+				User:  "root",
+				Host:  "localhost",
+				Query: "SHOW GRANTS FOR tester@localhost;",
+				Expected: []sql.Row{
+					{"GRANT SELECT, INSERT ON *.* TO `tester`@`localhost`"},
+				},
+			},
+		},
+	},
+	{
+		Name: "show grants at various scopes",
+		SetUpScript: []string{
+			"CREATE USER tester@localhost;",
+			"GRANT SELECT ON *.* to tester@localhost",
+			"GRANT SELECT ON db.* to tester@localhost",
+			"GRANT SELECT ON db.tbl to tester@localhost",
+		},
+		Assertions: []UserPrivilegeTestAssertion{
+			{
+				User:  "root",
+				Host:  "localhost",
+				Query: "SHOW GRANTS FOR tester@localhost;",
+				Expected: []sql.Row{
+					{"GRANT SELECT ON *.* TO `tester`@`localhost`"},
+					{"GRANT SELECT ON `db`.* TO `tester`@`localhost`"},
+					{"GRANT SELECT ON `db`.`tbl` TO `tester`@`localhost`"},
```

Comment:
Does this actually show all 3 in MySQL? Or does it just show `*.*` since that also implies the others? This question applies to the other tests as well.

Reply:
Yeah, even if they are redundant MySQL will show all of them

```
MySQL> show grants for new_user;
+-------------------------------------------------+
| Grants for new_user@%                           |
+-------------------------------------------------+
| GRANT SELECT ON *.* TO `new_user`@`%`           |
| GRANT SELECT ON `test_db`.* TO `new_user`@`%`   |
| GRANT SELECT ON `test_db`.`t` TO `new_user`@`%` |
+-------------------------------------------------+
```